src/messaging/Message.utils.ts

```diff
@@ -0,0 +1,297 @@
+import { CompressedAttestedClaim } from '../types/AttestedClaim'
+import {
+  QuoteUtils,
+  ClaimUtils,
+  AttestedClaimUtils,
+  MessageBody,
+  AttestationUtils,
+  MessageBodyType,
+  IAttestedClaim,
+  RequestForAttestationUtils,
+} from '..'
+import ITerms, { CompressedTerms } from '../types/Terms'
+import * as SDKErrors from '../errorhandling/SDKErrors'
+import {
+  CompressedRejectedTerms,
+  CompressedRequestAttestationForClaim,
+  CompressedSubmitAttestationForClaim,
+  CompressedRequestClaimsForCTypes,
+  CompressedRequestAcceptDelegation,
+  CompressedSubmitAcceptDelegation,
+  CompressedDelegationData,
+  CompressedInformCreateDelegation,
+  IRequestingClaimsForCTypes,
+  IPartialClaim,
+  IRequestingAttestationForClaim,
+  ISubmittingAttestationForClaim,
+  IRequestingAcceptDelegation,
+  ISubmitingAcceptDelegation,
+  IDelegationData,
+  IInformingCreateDelegation,
+  CompressedMessageBody,
+} from './Message'
+
+/**
+ * [STATIC] Compresses a [[Message]] depending on the message body type.
+ *
+ * @param body The body of the [[Message]] which depends on the [[MessageBodyType]] that needs to be compressed.
+ *
+ *
+ */
+
+export function compressMessage(body: MessageBody): CompressedMessageBody {
+  switch (body.type) {
+    case MessageBodyType.REQUEST_TERMS: {
+      const compressedContents = ClaimUtils.compress(body.content)
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.SUBMIT_TERMS: {
+      const compressedContents: CompressedTerms = [
+        ClaimUtils.compress(body.content.claim),
+        body.content.legitimations.map(
+          (attestedClaim: IAttestedClaim | CompressedAttestedClaim) =>
+            Array.isArray(attestedClaim)
+              ? attestedClaim
+              : AttestedClaimUtils.compress(attestedClaim)
+        ),
+        body.content.delegationId,
+        body.content.quote
+          ? QuoteUtils.compressAttesterSignedQuote(body.content.quote)
+          : undefined,
+        body.content.prerequisiteClaims,
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.REJECT_TERMS: {
+      const compressedContents: CompressedRejectedTerms = [
+        ClaimUtils.compress(body.content.claim),
+        body.content.legitimations.map((val) =>
+          AttestedClaimUtils.compress(val)
+        ),
+        body.content.delegationId ? body.content.delegationId : undefined,
+      ]
+      return [body.type, compressedContents]
+    }
+    case MessageBodyType.REQUEST_ATTESTATION_FOR_CLAIM: {
+      const compressedContents: CompressedRequestAttestationForClaim = [
+        RequestForAttestationUtils.compress(body.content.requestForAttestation),
+        body.content.quote
+          ? QuoteUtils.compressQuoteAgreement(body.content.quote)
+          : undefined,
+        body.content.prerequisiteClaims
+          ? body.content.prerequisiteClaims.map((claim) =>
+              ClaimUtils.compress(claim)
```

Comment:
_Info_: This line is currently not covered by tests.

Reply:
At the moment we don't use `prerequisiteClaims`. This might turn out to be removed, as it was part of the followup on the quotes. Let us speak with and see what we shall do about it.